### PR TITLE
 feat: Implement Spark-compatible CAST from Numeric to Binary

### PR DESCRIPTION
**Pull request by mattharder91**
## Which issue does this PR close?
https://github.com/apache/datafusion-comet/issues/405, https://github.com/apache/datafusion-comet/issues/377

Closes #.

### CODE REVIEW

File: spark/src/main/scala/org/apache/comet/expressions/CometCast.scala [hunk 1]
```diff
@@ -248,4 +250,12 @@ object CometCast {
     case _ => Unsupported
   }
 
+  private def canCastToBinary(toType: DataType): SupportLevel = toType match {
```
Comment (leoluan2009) [inline spark/src/main/scala/org/apache/comet/expressions/CometCast.scala:253]: fromType: DataType ?

Comment (mattharder91) [reply spark/src/main/scala/org/apache/comet/expressions/CometCast.scala]: thanks

**Comment by mattharder91** [comment]
I am getting 
```
Cannot resolve "TRY_CAST(a AS BINARY)" due to data type mismatch: cannot cast "BIGINT" to "BINARY"
```

In non ANSI Mode and respectfully for all other INT types. It is actually the behavior from spark  without comet which confuses me as I was expecting to work or at least return 0.

**Comment by viirya** [comment]
> ```
> Cannot resolve "TRY_CAST(a AS BINARY)" due to data type mismatch: cannot cast "BIGINT" to "BINARY"
> ```

Where did you get the error? Looks like BigInt to Binary is okay to cast.

**Comment by andygrove** [comment]
@mattharder91  I will close this PR for now since it has been inactive for a while. Feel free to reopen this or create a new PR if you continue work on this.

**Pull request closed without merging**